src/v/cluster/health_monitor_backend.cc

```diff
@@ -271,6 +277,271 @@ void health_monitor_backend::abort_current_refresh() {
     }
 }
 
+namespace {
+void record_risks_in_report(
+  restart_risk_report& report,
+  model::node_id self,
+  const followers_stats& fs,
+  const model::topic_namespace& nt,
+  model::partition_id pid) {
+    // 0 or 1 each
+    uint16_t self_out_of_sync = std::ranges::count(fs.out_of_sync, self);
+    uint16_t self_down = std::ranges::count(fs.down, self);
+    uint16_t self_in_sync = 1 - self_out_of_sync - self_down;
```

Comment:
Would be nice to protect against wrapping here. Maybe assert is too strict, but an error log message if both self_out_of_sync and self_down are true seems good.

Reply:
It looks like this comment is not addressed

Reply:
If they are both true (but each is just 1) the line below will fail because 65535 is not 0 nor 1. For it to give a false positive fs.down and fs.out_of_sync will need to contain the current node 65536 or 65537 times, which is fairly unlikely even if we have a bug that allows them to have it more than once.